ci/which-channel.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# This script outputs the channel where a CI workflow wants to merge into.
+#
+# Inputs:
+#     BASE_SHA    The commit SHA of the branch where the PR wants to merge into.
+#
+# GitHub Action Outputs:
+#     CHANNEL     Target channel where the PR wants to merge into.
+
+set -euo pipefail
+
+# When `BASE_SHA` is missing, we assume it is from bors merge commit,
+# so hope `HEAD~` to find the previous commit on master branch.
+base_sha=$(git rev-parse "${BASE_SHA:-HEAD~1}")
+
+# Get symbolic names for the base_sha.
+# Assumption: Cargo branches are always in the format of `rust-1.*.0`,
+#             otherwise `git name-rev` will return "undefined".
+ref=$(git name-rev --name-only --refs='origin/rust-1.*.0' $base_sha)
+
+# Get the latest `rust-1.*.0` branch from remote origin.
+# Assumption: The latest branch is always beta branch.
+beta=$(git branch --remotes --list 'origin/rust-1.*.0' | sort | tail -n1 | tr -d "[:space:]")
```

Comment:
This looks like it might have some issues with three-digit version like 1.100. Perhaps it could be something like `sed -E 's/ *upstream\/rust-1.([0-9]+).0/\1/g' | sort -n`?

Reply:
I went using `sort --version-sort`. Tested on ubuntu-latest (22.04). It can correctly sort the following versions from

```
origin/rust-1.100.0
origin/rust-1.50.0
origin/rust-1.222.0
origin/rust-1.48.0
origin/rust-1.70.0
origin/rust-1.60.0
origin/rust-1.69.0
origin/rust-1.9.0
```
to

```
origin/rust-1.9.0
origin/rust-1.48.0
origin/rust-1.50.0
origin/rust-1.60.0
origin/rust-1.69.0
origin/rust-1.70.0
origin/rust-1.100.0
origin/rust-1.222.0
```